ArtOfIllusion/src/artofillusion/animation/PoseTrack.java

```diff
@@ -85,8 +85,10 @@ public Track duplicate(Object obj)
     t.smoothingMethod = smoothingMethod;
     t.tc = tc.duplicate((ObjectInfo) obj);
     t.relative = relative;
-    t.theWeight = (WeightTrack) theWeight.duplicate(t);
-    t.subtracks = new Track [] {theWeight};
+    t.theWeight = theWeight.duplicate(t);
+    t.subtracks = new Track[this.subtracks.length];
+    for (int i = 0; i < this.subtracks.length; i++)
+      t.subtracks[i] = this.subtracks[i].duplicate(t);
```

Comment:
Perfect place for an 'enhanced for' 

`for (Track: this.subtracks)`

Reply:
I cannot use enhanced for becase i still need  i-index variable to put duplucates to target array.

Reply:
That's an annoying hole in the enhanced loop...